plugins/apache-airflow/src/api/ApacheAirflowClient.ts

```diff
@@ -97,11 +98,55 @@ export class ApacheAirflowClient implements ApacheAirflowApi {
   async updateDag(dagId: string, isPaused: boolean): Promise<Dag> {
     const init = {
       method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
       body: JSON.stringify({ is_paused: isPaused }),
     };
     return await this.fetch<Dag>(`/dags/${dagId}`, init);
   }
 
+  /**
+   * Get the latest DAG Runs of a specific DAG.
+   * The DAG runs are ordered by start date
+   *
+   * @remarks
+   *
+   * The "limit" option means the maximum number of DAG runs to return
+   * "objectsPerRequest" means the maximum number of DAG runs to be taken per fetch request.
+   * This is just to make sure the response payloads are not too big
+   */
+  async getDagRuns(
+    dagId: string,
+    options = { objectsPerRequest: 100, limit: 5 },
+  ): Promise<DagRun[]> {
+    const dagRuns: DagRun[] = [];
+    const searchParams: ListDagsParams = {
+      limit: Math.min(options.objectsPerRequest || 100, options.limit || 5),
+      offset: 0,
+      order_by: '-start_date',
+    };
+
+    for (;;) {
+      const response = await this.fetch<{
+        dag_runs: DagRun[];
+        total_entries: number;
+      }>(`/dags/${dagId}/dagRuns?${qs.stringify(searchParams)}`);
+      dagRuns.push(...response.dag_runs);
+
+      if (
+        dagRuns.length >= response.total_entries ||
```

Comment:
feels like we need another break condition in here, as in when the source has no more data to give. Maybe when `response.dag_runs.length` or `response.total_entries` is zero or so?